File: plugin/Weather/src/main/java/Weather.java
import android.content.Context;
import com.aliucord.Http;
import com.aliucord.Utils;
import com.aliucord.annotations.AliucordPlugin;
import com.aliucord.api.CommandsAPI.CommandResult;
import com.aliucord.entities.MessageEmbedBuilder;
import com.aliucord.entities.Plugin;
import com.discord.api.commands.ApplicationCommandType;
import weather.WeatherResponse;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;

@AliucordPlugin
public class Weather extends Plugin {
    private static final String THERMOMETER_EMOJI = "\uD83C\uDF21";
    private static final String CLOUD_EMOJI = "☁️";
    private static final String HUMIDITY_EMOJI = "\uD83D\uDCA6";
    private static final String UV_INDEX_EMOJI = "\uD83D\uDD76️";
    private static final String WIND_EMOJI = "\uD83D\uDCA8";

    @Override
    public void start(Context context) {
        commands.registerCommand(
            "weather",
            "Get the weather for the current location, or a specific location",
            java.util.Collections.singletonList(
                Utils.createCommandOption(
                    ApplicationCommandType.STRING,
                    "location",
                    "The location to query"
                )
            ),
            ctx -> {
                String location = ctx.getString("location");
                WeatherResponse weather;
                
                try {
                    weather = Http.simpleJsonGet(
                        "https://wttr.in/" + (location != null ? location : "") + "?format=j1",
                        WeatherResponse.class
                    );
                } catch (Throwable throwable) {
                    logger.error(throwable);
                    return new CommandResult("Uh oh, failed to fetch weather data", null, false);
                }

                WeatherResponse.Condition condition = weather.current_condition.get(0);
                WeatherResponse.TextValue weatherDesc = condition.weatherDesc.get(0);
                WeatherResponse.NearestArea nearestArea = weather.nearest_area.get(0);
                String areaName = nearestArea.areaName.get(0).value;
                String country = nearestArea.country.get(0).value;
                String region = nearestArea.region.get(0).value;

                StringBuilder description = new StringBuilder();
                description.append("**Temp**:\n");
                description.append(THERMOMETER_EMOJI).append(" **Feels Like**: `").append(condition.FeelsLikeF).append("°F`  |  `").append(condition.FeelsLikeC).append("°C`\n");
                description.append(THERMOMETER_EMOJI).append(" **Temperature**: `").append(condition.temp_F).append("°F`  |  `").append(condition.temp_C).append("°C`\n");
                description.append("**Wind**:\n");
                description.append(WIND_EMOJI).append(" **Wind Direction**: `").append(condition.winddir16Point).append("`\n");
                description.append(WIND_EMOJI).append(" **Wind Speed**: `").append(condition.windspeedMiles).append(" MPH`  |  `").append(condition.windspeedKmph).append(" KMPH`\n");
                description.append("**Other**:\n");
                description.append(CLOUD_EMOJI).append("  **Cloud Cover**: `").append(condition.cloudcover).append("%`\n");
                description.append(HUMIDITY_EMOJI).append(" **Humidity**: `").append(condition.humidity).append("`\n");
                description.append(UV_INDEX_EMOJI).append(" **UV Index**: `").append(condition.uvIndex).append("`\n");

                MessageEmbedBuilder embed = new MessageEmbedBuilder()
                    .setTitle("Weather: " + weatherDesc.value)
                    .setColor(0xEDEDED)
                    .setDescription(description.toString())
                    .setFooter(areaName + ", " + region + ", " + country, null);

                if (location != null) {
                    try {
                        embed.setUrl("http://wttr.in/" + URLEncoder.encode(location, "UTF-8"));
                    } catch (UnsupportedEncodingException e) {
                        logger.error(e);
                    }
                } else {
                    embed.setUrl("http://wttr.in/");
                }

                return new CommandResult(null, java.util.Collections.singletonList(embed.build()), false);
            }
        );
    }

    @Override
    public void stop(Context context) {
        commands.unregisterAll();
    }
}

File: plugin/Weather/src/main/kotlin/Weather.kt
import android.content.Context
import com.aliucord.Http
import com.aliucord.Utils
import com.aliucord.annotations.AliucordPlugin
import com.aliucord.api.CommandsAPI.CommandResult
import com.aliucord.entities.MessageEmbedBuilder
import com.aliucord.entities.Plugin
import com.discord.api.commands.ApplicationCommandType
import weather.WeatherResponse
import java.io.UnsupportedEncodingException
import java.net.URLEncoder

@AliucordPlugin
class Weather : Plugin() {
    companion object {
        private const val thermometerEmoji = "\uD83C\uDF21"
        private const val cloudEmoji = "☁️"
        private const val humidityEmoji = "\uD83D\uDCA6"
        private const val uvIndexEmoji = "\uD83D\uDD76️"
        private const val windEmoji = "\uD83D\uDCA8"
    }

    override fun start(context: Context) {
        val arguments =
            listOf(
                Utils.createCommandOption(
                    ApplicationCommandType.STRING,
                    "location",
                    "The location to query"
                )
            )

        commands.registerCommand(
            "weather",
            "Get the weather for the current location, or a specific location",
            arguments
        ) {
            val location = it.getString("location")
            val weather = try {
                Http.simpleJsonGet(
                    "https://wttr.in/${location.orEmpty()}?format=j1",
                    WeatherResponse::class.java
                )
            } catch (throwable: Throwable) {
                logger.error(throwable)
                return@registerCommand CommandResult(
                    "Uh oh, failed to fetch weather data",
                    null,
                    false
                )
            }

            val condition = weather.current_condition.first()
            val weatherDesc = condition.weatherDesc.first()
            val nearestArea = weather.nearest_area.first()
            val areaName = nearestArea.areaName.first().value
            val country = nearestArea.country.first().value
            val region = nearestArea.region.first().value

            val embed = MessageEmbedBuilder()
                .setTitle("Weather: ${weatherDesc.value}")
                .setColor(0xEDEDED)
                .setDescription(
                    buildString {
                        appendLine("**Temp**:")
                        appendLine(
                            "$thermometerEmoji **Feels Like**: `${condition.FeelsLikeF}°F`  |  `${condition.FeelsLikeC}°C`"
                        )
                        appendLine(
                            "$thermometerEmoji **Temperature**: `${condition.temp_F}°F`  |  `${condition.temp_C}°C`"
                        )
                        appendLine("**Wind**:")
                        appendLine("$windEmoji **Wind Direction**: `${condition.winddir16Point}`")
                        appendLine(
                            "$windEmoji **Wind Speed**: `${condition.windspeedMiles} MPH`  |  `${condition.windspeedKmph} KMPH`"
                        )
                        appendLine("**Other**:")
                        appendLine("$cloudEmoji  **Cloud Cover**: `${condition.cloudcover}%`")
                        appendLine("$humidityEmoji **Humidity**: `${condition.humidity}`")
                        appendLine("$uvIndexEmoji **UV Index**: `${condition.uvIndex}`")
                    }
                ).setFooter("$areaName, $region, $country", null)

            if (location != null) {
                try {
                    embed.setUrl("http://wttr.in/${URLEncoder.encode(location, "UTF-8")}")
                } catch (e: UnsupportedEncodingException) {
                    logger.error(e)
                }
            } else {
                embed.setUrl("http://wttr.in/")
            }

            CommandResult(null, listOf(embed.build()), false)
        }
    }

    override fun stop(context: Context) = commands.unregisterAll()
}